server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala

```diff
@@ -8202,48 +8305,74 @@ trait MailboxSetMethodContract {
       .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
       .body(request)
     .when
       .post
     .`then`
       .log().ifValidationFails()
       .statusCode(SC_OK)
-      .contentType(JSON)
-      .extract
-      .body
-      .asString
+      .body("methodResponses[0][1].updated", hasKey(mailboxId.serialize))
+  }
 
-    assertThatJson(response)
-      .whenIgnoringPaths("methodResponses[0][1].oldState", "methodResponses[0][1].newState")
-      .isEqualTo(
-      s"""{
-         |  "sessionState": "${SESSION_STATE.value}",
-         |  "methodResponses": [
-         |    ["Mailbox/set", {
-         |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
-         |      "notUpdated": {
-         |        "${mailboxId.serialize}": {
-         |          "type": "notFound",
-         |          "description": "#private:andre@domain.tld:mailbox"
-         |        }
-         |      }
-         |    }, "c2"]
-         |  ]
-         |}""".stripMargin)
+  @Test
+  def updateShouldFailWhenUpdateParentIdFromSharedMailboxAndDoesNotHasDeleteMailboxRight(server: GuiceJamesServer): Unit = {
+    val mailboxProbe = server.getProbe(classOf[MailboxProbeImpl])
+    val path = MailboxPath.forUser(ANDRE, "mailbox")
+    val mailboxId: MailboxId = mailboxProbe.createMailbox(path)
+    val parentPath = MailboxPath.forUser(BOB, "parent")
+    val parentId = mailboxProbe.createMailbox(parentPath)
+
+    val aCLProbeImpl = server.getProbe(classOf[ACLProbeImpl])
+    aCLProbeImpl.replaceRights(path, BOB.asString, MailboxACL.FULL_RIGHTS.except(new MailboxACL.Rfc4314Rights(Right.DeleteMailbox)))
+
+    val request = s"""
+                     |{
+                     |   "using": [ "urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail" ],
+                     |   "methodCalls": [
+                     |       [
+                     |           "Mailbox/set",
+                     |           {
+                     |                "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+                     |                "update": {
+                     |                    "${mailboxId.serialize}": {
+                     |                      "parentId": "${parentId.serialize}"
+                     |                    }
+                     |                }
+                     |           },
+                     |           "c2"
+                     |       ]
+                     |   ]
+                     |}
+                     |""".stripMargin
+
+    `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(SC_OK)
+      .body("methodResponses[0][1].notUpdated", hasKey(mailboxId.serialize))
+      .body("methodResponses[0][1].notUpdated." + mailboxId.serialize,
+        jsonEquals(s"""{
+                      |  "type": "invalidArguments",
```

Comment:
Should we take a chance to review the response error for this missing rights case?

IMO we should return the `forbidden` SetError instead of `invalidArguments`.
```
forbidden: The method and arguments are valid, but executing the method would violate an Access Control List (ACL) or other permissions policy.
```
cf https://jmap.io/spec-core.html

Reply:
I also had this concern when implemented,
but the code looks like this is not a forget

![image](https://github.com/user-attachments/assets/9dd3b3ab-50a6-4343-a01c-c6eafee984f6)

I asked myself, "Should 'forbirdden' be YES/NO for 'updated shared mailbox' instead of 'parentId'?"